src/server/RedirectingHttpHandler.ts

```diff
@@ -0,0 +1,106 @@
+import type { TargetExtractor } from '../http/input/identifier/TargetExtractor';
+import { RedirectResponseDescription } from '../http/output/response/RedirectResponseDescription';
+import type { ResponseWriter } from '../http/output/ResponseWriter';
+import { getLoggerFor } from '../logging/LogUtil';
+import { FoundHttpError } from '../util/errors/FoundHttpError';
+import { MovedPermanentlyHttpError } from '../util/errors/MovedPermanentlyHttpError';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import { PermanentRedirectHttpError } from '../util/errors/PermanentRedirectHttpError';
+import type { RedirectHttpError } from '../util/errors/RedirectHttpError';
+import { SeeOtherHttpError } from '../util/errors/SeeOtherHttpError';
+import { TemporaryRedirectHttpError } from '../util/errors/TemporaryRedirectHttpError';
+import { getRelativeUrl, joinUrl } from '../util/PathUtil';
+import type { HttpHandlerInput } from './HttpHandler';
+import { HttpHandler } from './HttpHandler';
+import type { HttpRequest } from './HttpRequest';
+
+const redirectErrorFactories: Record<301 | 302 | 303 | 307 | 308, (location: string) => RedirectHttpError> = {
+  301: (location: string): RedirectHttpError => new MovedPermanentlyHttpError(location),
+  302: (location: string): RedirectHttpError => new FoundHttpError(location),
+  303: (location: string): RedirectHttpError => new SeeOtherHttpError(location),
+  307: (location: string): RedirectHttpError => new TemporaryRedirectHttpError(location),
+  308: (location: string): RedirectHttpError => new PermanentRedirectHttpError(location),
+};
+
+/**
+ * Handler that redirects paths matching given patterns
+ * to their corresponding URL, substituting selected groups.
+ */
+export class RedirectingHttpHandler extends HttpHandler {
+  private readonly logger = getLoggerFor(this);
+  private readonly redirects: {
+    regex: RegExp;
+    redirectPattern: string;
+  }[];
+
+  /**
+   * Creates a handler for the provided redirects.
+   * @param redirects - A mapping between URL patterns.
```

Comment:
Not all params here are listed due to the changes. Either all or none is fine for me.